java-frontend/src/main/java/org/sonar/java/se/xproc/ExceptionalCheckBasedYield.java

```diff
@@ -138,6 +148,28 @@ public int hashCode() {
       .hashCode();
   }
 
+  public Set<List<JavaFileScannerContext.Location>> exceptionFlows() {
+    Set<List<JavaFileScannerContext.Location>> flows = FlowComputation.flow(node.parent(), svCausingException, domains(node.programState.getConstraints(svCausingException)));
+    Tree syntaxTree = node.programPoint.syntaxTree();
+    ImmutableSet.Builder<List<JavaFileScannerContext.Location>> flowBuilder = ImmutableSet.builder();
+
+    for (List<JavaFileScannerContext.Location> flow : flows) {
+      List<JavaFileScannerContext.Location> newFlow = ImmutableList.<JavaFileScannerContext.Location>builder()
+        .add(new JavaFileScannerContext.Location(exceptionType().name() + " is thrown.", syntaxTree))
```

Comment:
Cannot happen, exception has to be provided to create a `ExceptionalCheckBasedYield`.

I don't know if I should override the method and annotate it with `@NonNull`, but I will then have an issue from [S2637](https://jira.sonarsource.com/browse/RSPEC-2637) saying that the method is nonnull and returning null if I call super.exceptionType() within its body.